lib/Runtime/Language/JavascriptExceptionOperators.cpp

```diff
@@ -93,20 +93,22 @@ namespace Js
     {
         void *continuation = nullptr;
         JavascriptExceptionObject *exception = nullptr;
-
+        void *tryCatchFrameAddr = nullptr;
         PROBE_STACK(scriptContext, Constants::MinStackDefault + spillSize + argsSize);
-        Js::JavascriptExceptionOperators::TryCatchFrameAddrStack tryCatchFrameAddrStack(scriptContext, frame);
-
-        try
-        {
-            Js::JavascriptExceptionOperators::AutoCatchHandlerExists autoCatchHandlerExists(scriptContext);
-            continuation = amd64_CallWithFakeFrame(tryAddr, frame, spillSize, argsSize);
-        }
-        catch (const Js::JavascriptException& err)
         {
-            exception = err.GetAndClear();
-        }
+            Js::JavascriptExceptionOperators::TryCatchFrameAddrStack tryCatchFrameAddrStack(scriptContext, frame);
 
+            try
+            {
+                Js::JavascriptExceptionOperators::AutoCatchHandlerExists autoCatchHandlerExists(scriptContext);
+                continuation = amd64_CallWithFakeFrame(tryAddr, frame, spillSize, argsSize);
+            }
+            catch (const Js::JavascriptException& err)
+            {
+                exception = err.GetAndClear();
+                tryCatchFrameAddr = scriptContext->GetThreadContext()->GetTryCatchFrameAddr();
```

Comment:
>tryCatchFrameAddr [](start = 16, length = 17)

In what scenario this address be different from frame?

Reply:
It will be the same, but we save it on the threadcontext to retrieve it in OP_TryFinally. Added assert.